clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java

```diff
@@ -289,6 +430,7 @@ private void releasePool(String poolType, String poolName, ResourcePageSource re
     ServerSideConfiguration.Pool pool = resourcePageSource.getPool();
     OffHeapResource source = services.getService(OffHeapResourceIdentifier.identifier(pool.getServerResource()));
     if (source != null) {
+      unRegisterPoolStatistics(resourcePageSource);
```

Comment:
You need to make sure the `EhcacheActiveEntity.destroy` method triggers this as well.